runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -10570,6 +10570,274 @@ inlineCompareAndSwapNative(
    }
 
 
+/**
+ * \brief
+ *   Generate inlined instructions equivalent to java/lang/StringCoding.hasNegatives or java/lang/StringCoding.countPositives
+ *
+ * \param node
+ *   The tree node
+ *
+ * \param cg
+ *   The Code Generator
+ */
+static TR::Register* inlineHasNegativesOrCountPositives(TR::Node* node, TR::RecognizedMethod recognizedMethod, TR::CodeGenerator* cg)
+   {
+   // Arguments to hasNegatives or countPositives
+   // Byte array
+   TR::Register *bufReg = cg->evaluate(node->getChild(0));
+   // Offset (i.e. index to begin counting from)
+   TR::Register *offsetReg = cg->evaluate(node->getChild(1));
+   // Length of byte array
+   TR::Register *lengthReg = cg->evaluate(node->getChild(2));
+
+   TR::Register *loopLimitReg = cg->allocateRegister();
+   TR::Register *limitReg = cg->allocateRegister();
+   TR::Register *maskReg = cg->allocateRegister();
+   TR::Register *indexReg = cg->allocateRegister();
+   TR::Register *xmmChunkReg = cg->allocateRegister(TR_VRF);
+   TR::Register *chunkReg = cg->allocateRegister();
+
+   TR::RegisterDependencyConditions *dependencies = generateRegisterDependencyConditions((uint8_t)7, (uint8_t)7, cg);
```

Comment:
You need to call `stopAddingConditions()` to "close" the dependency list.

Also, if you don't plan on adding anymore deps to this after this point you should just create it with 6 deps rather than 7.